src/Compilers/CSharp/Test/Symbol/Symbols/PartialPropertiesTests.cs

```diff
@@ -4353,14 +4640,86 @@ static void Main()
                 Diagnostic(ErrorCode.WRN_PartialPropertySignatureDifference, "this").WithArguments("int C.this[int p1]", "int C.this[int p2]").WithLocation(6, 24));
 
-            var indexer = ((CSharpCompilation)verifier.Compilation).GetMember<NamedTypeSymbol>("C").Indexers.Single();
-            Assert.Equal("p1", indexer.Parameters.Single().Name);
+            void verify(ModuleSymbol module)
+            {
+                var indexer = module.GlobalNamespace.GetMember<NamedTypeSymbol>("C").Indexers.Single();
+                Assert.Equal("p1", indexer.Parameters.Single().Name);
+            }
+        }
+
+        /// <remarks>See also <cref name="DocumentationCommentCompilerTests.PartialIndexer_Paramref_03"/></remarks>
+        [Fact]
+        public void IndexerParameterNameDifference_Attributes()
+        {
+            var source = """
+                using System;
+
+                [AttributeUsage(AttributeTargets.All, AllowMultiple = true)]
+                public class Attr(string s) : Attribute { } // 1 (unread parameter)
+
+                partial class C
+                {
+                    [Attr(nameof(p1))]
+                    [Attr(nameof(p2))] // 2
+                    public partial int this[int p1]
+                    {
+                        get;
+
+                        [param: Attr(nameof(p1))] // 3
+                        [param: Attr(nameof(p2))]
+                        set;
+                    }
+
+                    [Attr(nameof(p1))]
+                    [Attr(nameof(p2))] // 4
+                    public partial int this[int p2] // 5
+                    {
+                        get => p2;
+
+                        [param: Attr(nameof(p1))] // 6
+                        [param: Attr(nameof(p2))]
+                        set => p2.ToString();
+                    }
+
+                    [Attr(nameof(p1))]
+                    [Attr(nameof(p2))] // 7
+                    public partial void M(int p1);
+                    public partial void M(int p2) { } // 8
+                }
+                """;
+
+            var verifier = CreateCompilation(source);
+            verifier.VerifyDiagnostics(
+                // (4,26): warning CS9113: Parameter 's' is unread.
+                // public class Attr(string s) : Attribute { } // 1 (unread parameter)
+                Diagnostic(ErrorCode.WRN_UnreadPrimaryConstructorParameter, "s").WithArguments("s").WithLocation(4, 26),
+                // (9,18): error CS0103: The name 'p2' does not exist in the current context
+                //     [Attr(nameof(p2))] // 2
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "p2").WithArguments("p2").WithLocation(9, 18),
+                // (14,29): error CS0103: The name 'p1' does not exist in the current context
+                //         [param: Attr(nameof(p1))] // 3
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "p1").WithArguments("p1").WithLocation(14, 29),
```

Comment:
I know why this is happening. The implementation part owns binding attributes on parameters.

https://github.com/dotnet/csharplang/blob/main/proposals/csharp-11.0/extended-nameof-scope.md doesn't appear to spec what should happen here.

I am hesitant to even spec the behavior here (which is already shipped for methods).